src/blocks/libvirt.rs

```diff
@@ -73,51 +73,95 @@ impl ConfigBlock for Libvirt {
 
 impl Block for Libvirt {
     fn update(&mut self) -> Result<Option<Update>> {
-        if !self
+        // Check if the connection is still active and re-instatiate it if it's not
+        match self
             .qemu_conn
             .is_alive()
-            .expect("unrecoverable error with qemu object")
+            .block_error("virt", "error when getting connection status to libvirt")
         {
-            self.qemu_conn = Connect::open_read_only(
-                &self
-                    .qemu_conn
-                    .get_uri()
-                    .expect("could not get URI from the currently connected QEMU object"),
-            )
-            .expect("could not re-connect to libvirtd")
+            Ok(true) => {}
+            Ok(false) => {
+                self.qemu_conn = match Connect::open_read_only(
+                    &self
+                        .qemu_conn
+                        .get_uri()
+                        .block_error("virt", "error in retrieving URI information from libvirt")
+                        .unwrap(), /* Only happens if memory corruption; then we should panic anyway */
+                )
+                .block_error("virt", "error in reconnecting to libvirt socket")
+                {
+                    Ok(c) => c,
+                    Err(e) => return Err(e),
+                };
+            }
+            Err(e) => {
+                self.text.set_state(State::Critical);
```

Comment:
That's not necessary. When the block returns an error, the bar will switch to "error mode" and just display the error (with critical state's theme).

The difference between `.block_error()` (defined [here](https://github.com/greshake/i3status-rust/blob/b7e4ff2a47031be023c785768041c057a42fac4c/src/errors.rs#L12)) and `.expect()` is that `.expect()` just panics and user can only see something like "status command existed unexpectedly", while `.block_error()` wraps the error with a (hopefully) useful message which you then propagate, which will be eventually printed to the user.

There are plans to improve error handling (see #1371) (actually, it _is_ implemented, but on [a separate branch](https://github.com/greshake/i3status-rust/tree/async)).